### PR TITLE
Adding Staticcheck for Interoperator codebase

### PR DESCRIPTION
**Pull request by subhankarc**

### CODE REVIEW

File: interoperator/pkg/client/clientset/versioned/clientset.go [hunk 1]
```diff
@@ -66,7 +66,7 @@ func NewForConfig(c *rest.Config) (*Clientset, error) {
 	configShallowCopy := *c
 	if configShallowCopy.RateLimiter == nil && configShallowCopy.QPS > 0 {
 		if configShallowCopy.Burst <= 0 {
-			return nil, fmt.Errorf("Burst is required to be greater than 0 when RateLimiter is not set and QPS is set to greater than 0")
```
Comment (vivekzhere) [inline interoperator/pkg/client/clientset/versioned/clientset.go:69]: This is generated code. WIll be overwritten, next time we run 
```
bash vendor/k8s.io/code-generator/generate-groups.sh client github.com/cloudfoundry-incubator/service-fabrik-broker/interoperator/pkg/client github.com/cloudfoundry-incubator/service-fabrik-broker/interoperator/api "osb:v1alpha1 resource:v1alpha1" --go-header-file hack/boilerplate.go.txt
```

File: interoperator/pkg/watches/watches_test.go [hunk 2]
```diff
@@ -31,8 +30,6 @@ var kubeConfig *rest.Config
 var sch *runtime.Scheme
 var c client.Client
 
-const timeout = time.Second * 5
```
Comment (vivekzhere) [inline interoperator/pkg/watches/watches_test.go:34]: Surprised to see so many unused variables. I always thought go by default points this out when compiling.

File: interoperator/pkg/client/clientset/versioned/clientset.go (hunk 1 above)
Comment (vivekzhere) [reply interoperator/pkg/client/clientset/versioned/clientset.go:69]: I tried this out. Somehow it is not getting overwritten.

Comment (subhankarc) [reply interoperator/pkg/client/clientset/versioned/clientset.go:69]: Yeah, I'll keep this then.

File: interoperator/pkg/watches/watches_test.go (hunk 2 above)
Comment (subhankarc) [reply interoperator/pkg/watches/watches_test.go:34]: yes! this was surprising!

**Pull request merged**